src/thunderbots/software/network_input/main.cpp

```diff
@@ -83,55 +87,91 @@ int main(int argc, char** argv)
     // Init our backend class
     Backend backend = Backend();
 
+    // Store the state of the most up to date World message
+    thunderbots_msgs::World world_msg;
+    // A map used to store the latest detection data for each camera
+    std::map<unsigned int, SSL_DetectionFrame> latest_detection_data;
+    // Stores the most recent geometry data received
+    SSL_GeometryData latest_geometry_data;
+
     // Main loop
     while (ros::ok())
     {
-        auto ssl_vision_packet_queue = ssl_vision_client->getVisionPacketQueue();
-        while (!ssl_vision_packet_queue.empty())
+        auto ssl_vision_packets = ssl_vision_client->getVisionPacketVector();
+        for (const auto& packet : ssl_vision_packets)
         {
-            auto ssl_vision_packet = ssl_vision_packet_queue.front();
-            ssl_vision_packet_queue.pop();
-
-            std::optional<thunderbots_msgs::Field> field_msg =
-                backend.getFieldMsg(ssl_vision_packet);
-            if (field_msg)
+            if (packet.has_geometry())
             {
-                field_publisher.publish(*field_msg);
+                latest_geometry_data = packet.geometry();
             }
 
-            std::optional<thunderbots_msgs::Ball> ball_msg =
-                backend.getFilteredBallMsg(ssl_vision_packet);
-            if (ball_msg)
+            if (packet.has_detection())
             {
-                ball_publisher.publish(*ball_msg);
-            }
-
-            std::optional<thunderbots_msgs::Team> friendly_team_msg =
-                backend.getFilteredFriendlyTeamMsg(ssl_vision_packet);
-            if (friendly_team_msg)
-            {
-                friendly_team_publisher.publish(*friendly_team_msg);
+                auto detection = packet.detection();
+                // Add the detection to the map, replacing any existing values
+                auto ret = latest_detection_data.insert(
+                    std::make_pair(detection.camera_id(), detection));
+                if (!ret.second)
+                {
+                    ret.first->second = detection;
+                }
```

Comment:
I'm a bit confused about what `ret` is here and what we're doing with it?

Reply:
the return type of map::insert is a pair, where the first element is an iterator pointing to the object in the map, and the second is a boolean if the item was inserted. If the item was not inserted, it means the key already existed and the iterator points to that entry. So then we modify that entry to replace the item.

tl;dr used to add entries to the map and allow replacement of existing items

Reply:
Ok, could we add a comment or two just explaining that in this section? Just some really brief ones, like `// Check that we inserted succesfully` would be fine.

Reply:
fixed